### PR TITLE
use golang:alpine for golang docker image

### PR DESCRIPTION
**Pull request by vladbarosan**
Address https://github.com/Microsoft/vscode-docker/issues/176
- Use `golang:alpine` since `golang:onbuild` is deprecated and discouraged from being used.
- Use multi-stage build
- use `ca-certificates` to enable SSL certificate check

### CODE REVIEW

**Comment by vladbarosan** [comment]
@brettcannon can you have a look please, this addresses issue #176

**Comment by brettcannon** [comment]
@vladbarosan I'm definitely not qualified to review this since I don't code in Go. 😁

File: configureWorkspace/configure.ts [hunk 1]
```diff
@@ -20,29 +20,25 @@ CMD ${cmd}`;
...
-# and configures it to run on startup 
-FROM golang:onbuild
-LABEL Name=${serviceName} Version=${version} 
+# golang:alpine is based on Alpine Linux and as a result contains a very
+# slim base image. This dockerfile configures ssh, bash for remote access
+# and git for running go. Customize the Dockerfile as required.
+FROM golang:alpine
+WORKDIR /go/src/app
+COPY . .
+RUN apk update && apk upgrade && apk add --no-cache bash git openssh
```
Comment (joshgav) [inline configureWorkspace/configure.ts]: not all would agree with including SSH, see e.g. [link](https://blog.docker.com/2014/06/why-you-dont-need-to-run-sshd-in-docker/). my vote would be to leave it out.

**Comment by joshgav** [comment]
thanks @vladbarosan! would it be better to provide a multi-stage build? see e.g. [link](https://medium.com/travis-on-docker/multi-stage-docker-builds-for-creating-tiny-go-images-e0e1867efe5a). [Buffalo](https://github.com/gobuffalo/buffalo) creates a multistage build too, see <https://github.com/gobuffalo/buffalo/blob/master/generators/docker/multi/templates/Dockerfile.tmpl>

**Comment by joshgav** [comment]
cc @bketelsen @ramya-rao-a @erikstmartin

File: configureWorkspace/configure.ts (hunk 1 above)
Comment (bketelsen) [reply configureWorkspace/configure.ts]: There's no reason to include ssh in a generic builder.

**Comment by bketelsen** [comment]
Go containers are often serving TLS services, it would make sense to add `ca-certificates` to this dockerfile so that certificates can be verified.

**Comment by bketelsen** [comment]
Lgtm

**Comment by vladbarosan** [comment]
@brettcannon can we merge this please ? thanks

**Comment by brettcannon** [comment]
@vladbarosan I'm still the wrong person to ask this as I've never written a Dockerfile and I still don't code in Go.

File: configureWorkspace/configure.ts [hunk 2]
```diff
@@ -20,29 +20,28 @@ CMD ${cmd}`;
...
+COPY . .
+RUN apk add --no-cache git
+RUN go-wrapper download   # "go get -d -v ./..."
+RUN go-wrapper install    # "go install -v ./..."
+
+#final stage
+FROM alpine:latest
+RUN apk --no-cache add ca-certificates
+COPY --from=builder /go/bin/app /app
+ENTRYPOINT ./app
```
Comment (joshgav) [inline configureWorkspace/configure.ts]: should this be `/app`?

**Comment by chrisdias** [comment]
LGTM

**Pull request merged**